pkg/variablefinders/requirements.go

```diff
@@ -13,61 +13,21 @@ import (
 
 // FindRequirements finds the requirements from the dev Environment CRD
 func FindRequirements(g gitclient.Interface, jxClient jxc.Interface, ns, dir, owner, repo string) (*jxcore.RequirementsConfig, error) {
-	settings, err := requirements.LoadSettings(dir, true)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to load settings")
-	}
-	if settings == nil {
-		// lets use an empty settings file
-		settings = &jxcore.Settings{}
-	}
-
-	gitURL := ""
-	if settings != nil {
-		gitURL = settings.Spec.GitURL
-	}
-	if gitURL == "" {
-		if ns == "" {
-			ns = jxcore.DefaultNamespace
-		}
-		env, err := jxenv.GetDevEnvironment(jxClient, ns)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to get dev environment")
-		}
-		if env == nil {
-			return nil, errors.Errorf("failed to find a dev environment source url as there is no 'dev' Environment resource in namespace %s", ns)
-		}
-		gitURL = env.Spec.Source.URL
-		if gitURL == "" {
-			return nil, errors.New("failed to find a dev environment source url on development environment resource")
-		}
-	}
-
 	// now lets merge the local requirements with the dev environment so that we can locally override things
 	// while inheriting common stuff
-	req, clusterDir, err := requirements.GetRequirementsAndGit(g, gitURL)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get requirements from cluster git repo")
-	}
-	if req == nil {
-		r := jxcore.NewRequirementsConfig()
-		req = &r.Spec
-	}
 
-	// lets see if we have organisation settings
-	srcConfig, err := sourceconfigs.LoadSourceConfig(clusterDir, true)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to load source configs")
-	}
-	groupSettings := sourceconfigs.FindSettings(srcConfig, owner, repo)
-
-	settings, err = mergeSettings(settings, groupSettings)
+	settings, clusterDir, err := GetSettings(g, jxClient, ns, dir, owner, repo)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to merge settings")
+		return nil, err
```

Comment:
can we do error.Wrap(err, "something like failed to get settings from ....")?
Might help some users to get more context around errors.

Reply:
OK